### PR TITLE
Fix `HeadersToKeep` and `MaxBridgedAuthorities` in Millau benchmarks

### PR DESCRIPTION
**Pull request by svyatonik**
1st point from https://github.com/paritytech/parity-bridges-common/issues/1543#issuecomment-1418814066

### CODE REVIEW

File: modules/grandpa/src/benchmarking.rs [hunk 1]
```diff
@@ -53,26 +53,29 @@ use sp_finality_grandpa::AuthorityId;
...
 	MAX_VOTE_ANCESTRIES_RANGE_BEGIN + MAX_VOTE_ANCESTRIES_RANGE_BEGIN;
 
+// the same with validators - if there are too much validators, let's run benchmarks on subrange
+fn validator_set_range_end<T: Config<I>, I: 'static>() -> u32 {
+	let max_bridged_authorities = T::MaxBridgedAuthorities::get();
+	if max_bridged_authorities > 128 {
+		max_bridged_authorities / 5
+	} else {
+		max_bridged_authorities
+	}
```
Comment (serban300) [inline modules/grandpa/src/benchmarking.rs:76]: Would it make sense to do something like

```
	if max_bridged_authorities > 128 {
		std::cmp::max(128, max_bridged_authorities / 5)
	}
```

?

File: modules/grandpa/src/lib.rs [hunk 2]
```diff
@@ -464,6 +467,20 @@ pub mod pallet {
...
+		T::HeadersToKeep::get()
+	}
+
+	/// Return number of headers to keep in the runtime storage.
+	#[cfg(feature = "runtime-benchmarks")]
+	pub(crate) fn headers_to_keep<T: Config<I>, I: 'static>() -> u32 {
+		// every db operation (significantly) slows down benchmarks, so let's keep as min as
+		// possible
+		2
+	}
```
Comment (serban300) [inline modules/grandpa/src/lib.rs:482]: Nit: Could we do this distinction in the mock runtime instead ? Is the mock runtime used for benchmarks ?

**Review by serban300 — APPROVED** [review]

File: modules/grandpa/src/benchmarking.rs (hunk 1 above)
Comment (svyatonik) [reply modules/grandpa/src/benchmarking.rs:76]: Yeah, seems like a good idea

File: modules/grandpa/src/lib.rs (hunk 2 above)
Comment (svyatonik) [reply modules/grandpa/src/lib.rs:482]: No, benchmarks are using the actual runtimes. That's why the weights of our pallets may be (are) different on Rialto, on Millau and on WBH/RBH. So every runtime generates its own `weights.rs`  file (e.g. for `frame-system` ; https://github.com/paritytech/polkadot/blob/master/runtime/rococo/src/weights/frame_system.rs, https://github.com/paritytech/substrate/blob/master/frame/system/src/weights.rs, https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/frame_system.rs)

**Review by acatangiu — APPROVED** [review]

**Pull request merged**